comprehensive/lms/static/cookieBanner.js

```diff
@@ -5,28 +5,64 @@ oxa = window.oxa || {};
 (function (namespace) {
     namespace.cookieBanner = cookieBanner;
     var proto = cookieBanner.prototype;
-    var consentCookieName = "cookie-banner";
+    proto.consentCookieName = "cookie-banner";
 
     // constructor
-    function cookieBanner() {
-        var self = this;
-
-        // track the user clicks, anchor/button and exclude the LearnMore link on the banner itself
-        $(document).ready(function () {
-            if (self.getCookie(consentCookieName) !== "true") {
-                $("a, button").on("click", proto.cookieClickHandler);
-            } else {
-                self.closeCookieBanner();
+    function cookieBanner() {};
+
+    // initialize the cookie consent api values
+    proto.init = function (cookieValues){
+        try {
+            if (proto.Error == null && cookieValues != null) {
+                // TODO:Check for each and every value availability, if missing value, just use our banner
+                proto.IsConsentRequired = cookieValues.IsConsentRequired;
+                proto.CookieName = cookieValues.CookieName;
+                proto.Css = cookieValues.Css[0];
+                proto.Domain = cookieValues.Domain;
+                proto.Js = cookieValues.Js[0];
+                proto.Locale = cookieValues.Locale;
+                proto.Markup = cookieValues.Markup;
+                proto.Error = cookieValues.Error;
+                proto.MinimumConsentDate = new Date(cookieValues.MinimumConsentDate).getTime();
             }
+        }
+        catch(error) {
+        }
+    };
+   
+    // Loads the Cookie API JS
+    // cache:false, causes the timestamp parameters to be added on the js, where the service is not accepting currently
+    proto.LoadJSCookieAPI = function (url, options) {
+        // Allow user to set any option except for dataType, cache, and url
+        options = $.extend(options || {}, {
+            dataType: "script",
+            cache: true,
+            url: url
         });
+        return jQuery.ajax(options);
     };
 
-    // Click handler for the cookie consent
-    proto.cookieClickHandler = function () {
-        if (this.id !== "btnPrivacy") {
-            proto.closeCookieBanner();
-        }
+   // Adds function BI cookies
+    proto.addBICookies = function () {
+        // Allow BI cookies
+        (function (a, b, c, d) {
```

Comment:
i'm blocking based on these variables names

Reply:
that script is given to us from the analytics team, I would prefer to keep them as is given

Reply:
okay. i agree we shouldn't deviate in this case so that can more easily consume future updates.